text/0000-singleton-removal.md

```diff
@@ -0,0 +1,69 @@
+- Feature Name: `singleton-remove`
+- Start Date: 2021-02-11
+- RFC PR: [RFC#11](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/11)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This change removes the `NetworkingManager.Singleton` static, as well as the majority of other static state in MLAPI. 
+
+# Motivation
+[motivation]: #motivation
+
+The purpose of the change is to allow multiple instances of the MLAPI Networking Stack to co-exist in the same process. This enables:
+- in-process servers. This allows developers to architect single-player and host-mode versions of their multiplayer game identically to the dedicated-server version of the same game.
+- integration testing. Both games that use MLAPI and MLAPI itself can write full server/client tests that run entirely in the scope of a single process. 
+- clients talking to multiple servers simultaneously. This is expected to be a requirement for MMO support. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+MLAPI supports having multiple `NetworkingManager`s running concurrently. There are a few things to keep in mind when doing so:
+- The simplest case is to only have one `NetworkingManager` per scene. This is what you do to support an in-process server: load a "server" version of your scene, and then call `StartServer` on the `NetworkingManager` in it. All "server" physics calculations must then be performed via the `PhysicsScene` associated with the server scene.
```

Comment:
So let's say we have a chat connection.  You'd want that to be across all scenes.  Or, perhaps you'd have a chat scene?

Reply:
Or perhaps to your point in here, that's be a case where you roll your own - instantiate it yourself and don't use the NetManager convenience

Reply:
perhaps `NetInstance` since `NetManager` will no longer be a singleton and "the Manager" :)

Reply:
You could have a Chat NetworkingManager GameObject with the DontDestroyOnLoad flag. I don't think there's any reason you couldn't have two Server NetworkingManagers running simultaneously in the same scene.